designs/inter-ident-verify.md

```diff
@@ -3,76 +3,96 @@
+7. Corp recognizes Support as valid, so proceeds.
+8. Corp decrypts message with Support public key from server X.
+9. Corp decrypts result with Corp private key from server X.
+10. Corp decrypts result with User public key from message. Corp now has the original message.
+
+11. Corp encrypts result with Corp private key from server X, acting as a signature from Corp directly.
+12. Corp encrypts result with user public key from message, preventing Support from decrypting the original message.
+13. Corp encrypts result with Support public key from server X, verifying validity of Support.
+14. Corp sends result to Support.
+
+#### Trusted caller (Support)
+15. Support decrypts message with Support private key from server Y, and can't decrypted further.
+16. Support sends result to User via phone call.
+
+#### Individual user (User)
+17. User decrypts result with User private key from server Y.
+18. User decrypts message with Corp public key from server Y.
+19. User verifies that result is the same as the original message generated.
+
+It's important to note that Support has no obligation to send User's public key to Corp, but then Corp would not be able to recover the original message, and the verification would fail because garbage in, garbage out. Therefore, the verification can only pass if Support sends a valid copy of User's public key to Corp. Following this algorithm, Support can never decrypt the message being sent between the User and Corp, and the Corp verifies and signs User's unique message. 
+
+The key aspect here is that the message being sent and received by User is the same, but encrypted in a different order. This means that it's essential to use a non-communitive cryptosystem.
 ___
 ## Readable hashes
-An important aspect of this process is that users are able to read their hashes over the phone. To increase the readability of these hashes, I propose a modified number representation system similar to hexadecimal where each digit has 64 variants, `{a-z, A-Z, 0-9, *, #}`, allowing 6-bits of information to be stored per digit. With only 6 digits in this number system, we can represent up to 68,719,476,736, covering all 32-bit numbers. With 11 digits, we can represent up to 7.38e19, covering all 64-bit numbers. For shorter messages, this should be enough to read these values over voice.
+An important aspect of this process is that users are able to read their hashes over the phone. To increase the readability of these hashes, I propose using English sentences to encode hashes. Since Ed25519 hashes create 512-bit signatures, we would need 36 words if we had a word bank with 20,000 words. If we followed a basic sentence structure to make these envoding more natural to read, where each sentence could be made up of 5 words, we could encode a hashing in ~7 simple sentences, which is incredible for the security we are ensuring.
+
+TODO: this requires 20,000 words for each object in the sentence, so not quite feasible?
```

Comment:
There is an easier way I think here. I think you could have a TOTP half-key on each users account, and when User A wants to verify user B, you concat their secrets together to get a full TOTP key that is a unique combination between the two individuals. Then you can simply display the TOTP and have the person read it.

For challenge response, You could have it so that User A + User B key is one direction, and User B  + User A key is the other since both will yield unique TOTPs. Then it's a 6 digit number that is continually changing, and is combined from User A + User B, without either user ever having the content of the secret exposed (Kani itself would arbitrate that you may read or see only your key muxed with someone elses).

This defends from say User C being able to impersonate User A, because Kani would only allow them to mix User C's key with the half key from A *or* B. They will never be able to see the factor that A and B both provide.